src/coreclr/System.Private.CoreLib/src/System/TypedReference.CoreCLR.cs

```diff
@@ -16,6 +17,21 @@ public ref partial struct TypedReference
         private readonly ref byte _value;
         private readonly IntPtr _type;
 
+        // implementation of CORINFO_HELP_GETREFANY
+        [StackTraceHidden]
+        internal static ref byte GetRefAny(IntPtr clsHnd, TypedReference typedByRef)
+        {
+            if (clsHnd != typedByRef._type)
+            {
+                ThrowInvalidCastException();
+            }
+
+            return ref typedByRef._value;
+
+            [DoesNotReturn]
+            static void ThrowInvalidCastException() => throw new InvalidCastException();
```

Comment:
This also needs `StackTraceHidden` afair.

Reply:
Good point
```suggestion
            [StackTraceHidden]
            static void ThrowInvalidCastException() => throw new InvalidCastException();
```